### PR TITLE
Pyic 8253 - Stored Identity Service Api Tests

### PR DESCRIPTION
**Pull request by undermad**
### What changed

API tests for stored identity service

### Why did it change

Storing a reusable identity allows returning users to access future services without re-proving their identity.  
By capturing the most recent VC bundle and identity state after key journeys, the system ensures that the SIS record remains accurate, valid, and aligned with the latest user information.  

### Issue tracking

- [PYIC-8253](https://govukverify.atlassian.net/browse/PYIC-8253)

## Checklists

- [x] API/unit/contract tests have been written/updated

[PYIC-8253]: https://govukverify.atlassian.net/browse/PYIC-8253?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by undermad** [comment]
Some p1 and p2 journes are covered in [P1](https://github.com/govuk-one-login/ipv-core-back/blob/main/api-tests/features/stored-identity/p1-journeys.feature) and [P2](https://github.com/govuk-one-login/ipv-core-back/blob/main/api-tests/features/stored-identity/p2-journeys.feature) files.

Very likely Ill need to add or change some tests, depends on the initial review.

File: api-tests/features/cimit/p1-alternate-doc.feature [hunk 1]
```diff
@@ -32,6 +32,7 @@ Feature: P1 CIMIT - Alternate doc
       Then I get an OAuth response
       When I use the OAuth response to get my identity
       Then I get a 'P1' identity
+      And I don't have a stored identity in EVCS
```
Comment (thebauSoftwire) [inline api-tests/features/cimit/p1-alternate-doc.feature]: I think it's fine to just add this for one or two key journeys e.g. a new identity journey since we're just checking that if the feature flag is turned off then we don't store an identity. It'll also make it easier when we turn on the feature flag as we won't have to update all of these tests

File: api-tests/features/stored-identity/cimit-journeys.feature [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+      Then I get a 'P2' identity
+      And I have a 'GPG45' stored identity record type with a 'P2' vot
+
+      Examples:
+        | document-details             |
+        | kenneth-passport-valid       |
+        | kenneth-driving-permit-valid |
+
+  Rule: P1 - V03 Enhanced Verification
+    Scenario Outline: Same session F2F enhanced verification mitigation
```
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/cimit-journeys.feature]: I think it might be good to also have separate session V03 mitigations - we could check that the stored identity wasn't created before they start a new session. To minimise the number of tests, we could just have one P1 test for this

File: api-tests/features/stored-identity/cimit-journeys.feature [hunk 3]
```diff
@@ -0,0 +1,247 @@
+Feature: Stored Identity Service - CIMIT journeys
+  Rule: P1 - D02 Mitigation
+    Background:
+      Given I activate the 'storedIdentityService' feature set
```
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/cimit-journeys.feature]: this could be moved to a top-level background section above all the rules to turn on the feature set

File: api-tests/features/stored-identity/m1c-journeys.feature [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      Then I get an 'address' CRI response
+      When I submit 'kenneth-current' details to the CRI stub
+      Then I get a 'fraud' CRI response
+      When I submit 'kenneth-unavailable' details to the CRI stub
+      Then I get a 'page-ipv-success' page response
+      When I submit a 'next' event
+      Then I get an OAuth response
+      When I use the OAuth response to get my identity
+      Then I get a 'P2' identity
+      And I have a 'GPG45' stored identity record type with a 'P2' vot
```
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/m1c-journeys.feature:28]: I think this is a duplicate of [this one in /stored-identity/p2-journeys.feature](https://github.com/govuk-one-login/ipv-core-back/blob/09fa4cba28d68b97705aa9da10a223e9bfdbb269/api-tests/features/stored-identity/p2-journeys.feature#L117-L132)? I think it might also be fine to just have that or just this test for new M1C profile (uk address) so we don't have too many duplicate tests

File: api-tests/features/stored-identity/m1c-journeys.feature [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+      Then I get a 'address' CRI response
+      When I submit 'kenneth-changed' details to the CRI stub
+      Then I get a 'fraud' CRI response
+      When I submit 'kenneth-unavailable' details to the CRI stub
+      Then I get a 'page-ipv-success' page response with context 'updateIdentity'
+      When I submit a 'next' event
+      Then I get an OAuth response
+      When I use the OAuth response to get my identity
+      Then I get a 'P2' identity
+      And I have a 'GPG45' stored identity record type with a 'P2' vot
```
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/m1c-journeys.feature]: To reduce duplicate tests it might also be fine to just have `Scenario Outline: Existing M1C address and name change - <selected-name-change>` or `Scenario Outline: Existing M1C name change - <selected-name-change>` and `Scenario: Existing M1C address change` especially as you've also added more update scenarios in reuse and rfc files :)

File: api-tests/features/stored-identity/p1-reuse-existing-identity.feature [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    And my identity 'GivenName' is '<expected-given-name>'
+    And my identity 'FamilyName' is '<expected-family-name>'
+    And I have a 'GPG45' stored identity record type with a 'P2' vot
+
+    Examples:
+      | selected-name-change | details                                          | fraud-details                       | expected-given-name | expected-family-name |
+      | family-name-only     | kenneth-changed-family-name-driving-permit-valid | kenneth-changed-family-name-score-2 | Kenneth             | Smith                |
+      | given-names-only     | kenneth-changed-given-name-driving-permit-valid  | kenneth-changed-given-name-score-2  | Ken                 | Decerqueira          |
+
+  Scenario Outline: Successful Address and Name Change - <selected-name-change>
```
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/p1-reuse-existing-identity.feature]: we could probably also remove this as it's very similar to the above two

File: api-tests/features/stored-identity/p1-repat-fraud-check.feature [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    And my identity 'GivenName' is '<expected-given-name>'
+    And my identity 'FamilyName' is '<expected-family-name>'
+    And I have a 'GPG45' stored identity record type with a 'P2' vot
+
+    Examples:
+      | selected-name-change | details                                          | fraud-details                       | expected-given-name | expected-family-name |
+      | given-names-only     | kenneth-changed-family-name-driving-permit-valid | kenneth-changed-family-name-score-2 | Kenneth             | Smith                |
+      | family-name-only     | kenneth-changed-given-name-driving-permit-valid  | kenneth-changed-given-name-score-2  | Ken                 | Decerqueira          |
+
+  Scenario Outline: Fraud 6 Months Expiry + Address and Name Change - <selected-name-change>
```
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/p1-repat-fraud-check.feature:26]: again, might be fine to just have this test or the ones above to reduce test duplicates

**Review by thebauSoftwire — COMMENTED** [review]
Nice test coverage! Just a couple of comments to potentially reduce test duplication

**Comment by thebauSoftwire** [comment]
do we also need some tests for the reprove journey?

File: api-tests/features/stored-identity/m1c-journeys.feature (hunk 4 above)
Comment (undermad) [reply api-tests/features/stored-identity/m1c-journeys.feature:28]: I deleted test from `p2-journeys.feature` and also moved another `No stored identity - unsuccessful M1C journey` to `m1c-journeys.feature` to keep all related tests in one place

File: api-tests/features/stored-identity/cimit-journeys.feature [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+      When I submit a 'next' event
+      Then I get an OAuth response
+      When I use the OAuth response to get my identity
+      Then I get a 'P1' identity
+      And I have a 'GPG45' stored identity record type with a 'P2' vot
+
+      Examples:
+        | document-details             |
+        | kenneth-passport-valid       |
+        | kenneth-driving-permit-valid |
```
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/cimit-journeys.feature]: this is technically a same-session mitigation journey as the next step on line 188 is starting the mitigation journey on the same session they received the CI :)

File: api-tests/features/stored-identity/cimit-journeys.feature [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+
+    Scenario Outline: Successful F2F enhanced verification mitigation - separate session
+      And I don't have a stored identity in EVCS
```
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/cimit-journeys.feature]: I don't think we need this line as we have the top-level background step on line 4

File: api-tests/features/stored-identity/cimit-journeys.feature (hunk 8 above)
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/cimit-journeys.feature]: it's probably fine to also just do one of these cases and replace this `Scenario Outline` with just a normal `Scenario`

File: api-tests/features/stored-identity/cimit-journeys.feature [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+      When I start new 'low-confidence' journeys until I get a 'page-ipv-reuse' page response
+      When I submit a 'next' event
+      Then I get an OAuth response
+      When I use the OAuth response to get my identity
+      Then I get a 'P1' identity
+
+      Examples:
+        | document-details             |
+        | kenneth-passport-valid       |
+        | kenneth-driving-permit-valid |
```
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/cimit-journeys.feature]: same here, probably fine to do just one of these cases

File: api-tests/features/stored-identity/p1-journeys.feature [hunk 11]
```diff
@@ -1,3 +1,4 @@
+@Build
```
Comment (thebauSoftwire) [inline api-tests/features/stored-identity/p1-journeys.feature:1]: nice!

**Review by thebauSoftwire — COMMENTED** [review]
Looking good! Just a couple of small comments

File: api-tests/features/stored-identity/cimit-journeys.feature (hunk 8 above)
Comment (undermad) [reply api-tests/features/stored-identity/cimit-journeys.feature]: Ah, my bad - it should be "same session" as separate session is below

**Review by thebauSoftwire — APPROVED** [review]
